atlasdb-config/src/test/java/com/palantir/atlasdb/factory/AtlasDbDialogueServiceProviderTest.java

```diff
@@ -0,0 +1,157 @@
+    private static final SslConfiguration SSL_CONFIGURATION
+            = SslConfiguration.of(Paths.get("var/security/trustStore.jks"));
+
+    private static final String CLIENT = "tom";
+    private static final String TIMESTAMP_PATH = "/tl/ts/" + CLIENT;
+    private static final MappingBuilder TIMESTAMP_MAPPING = post(urlEqualTo(TIMESTAMP_PATH));
+    private static final DialogueClients.ReloadingFactory DIALOGUE_BASE_FACTORY
+            = DialogueClients.create(Refreshable.only(ServicesConfigBlock.builder().build()));
+    private static final UserAgent USER_USER_AGENT = UserAgent.of(UserAgent.Agent.of("jeremy", "77.79.12"));
+
+    private int serverPort;
+    private ConjureTimelockService conjureTimelockService;
+
+    @Rule
+    public WireMockRule server = new WireMockRule(WireMockConfiguration.wireMockConfig().dynamicPort());
+
+    @Before
+    public void setup() {
+        setupServerToGiveOutTimestamps();
+
+        serverPort = server.port();
+        ServerListConfig serverListConfig = ImmutableServerListConfig.builder()
+                .addServers(getUriForPort(serverPort))
+                .sslConfiguration(SSL_CONFIGURATION)
+                .build();
+
+        AtlasDbDialogueServiceProvider provider = AtlasDbDialogueServiceProvider.create(
+                Refreshable.only(serverListConfig),
+                DIALOGUE_BASE_FACTORY,
+                USER_USER_AGENT);
+        conjureTimelockService = provider.getConjureTimelockService();
+    }
+
+
+    @Test
+    public void canMakeRequestsThroughDialogue() {
+        ConjureGetFreshTimestampsResponse response = makeTimestampsRequest();
+        assertThat(response.getInclusiveLower()).isEqualTo(58);
+        assertThat(response.getInclusiveUpper()).isEqualTo(70);
+    }
+
+    @Test
+    public void requestsAreIdentifiedWithTheUserProvidedUserAgent() {
+        makeTimestampsRequest();
+
+        server.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader("User-Agent", WireMock.containing(USER_USER_AGENT.primary().name())));
+        server.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader("User-Agent", WireMock.containing(USER_USER_AGENT.primary().version())));
+    }
+
+    @Test
+    public void atlasDbHttpClientVersionProvidedAsAnInformationalAgent() {
+        makeTimestampsRequest();
+
+        server.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader("User-Agent", WireMock.containing(
+                        String.format("%s/%s",
+                                AtlasDbRemotingConstants.ATLASDB_HTTP_CLIENT,
+                                AtlasDbRemotingConstants.CURRENT_CLIENT_PROTOCOL_VERSION.getProtocolVersionString()))));
+    }
+
+    @Test
+    public void resilientToRepeatedRedirects() {
+        server.stubFor(TIMESTAMP_MAPPING.willReturn(aResponse()
+                .withStatus(308)
+                .withHeader("Location", getUriForPort(serverPort))));
```

Comment:
Magic header strings are not the best, also for user agents

Reply:
Yep, updated to use `HttpHeaders`.